cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -13,9 +15,27 @@
 
 #include "SpCore/Assert.h"
 #include "SpCore/Log.h"
+#include "SpCore/Rpclib.h"
 #include "SpCore/Unreal.h"
 #include "SpEngine/EntryPointBinder.h"
 
+// We use MSGPACK macros here to define structs that can be passed into, and returned from, the service entry
+// points defined below. There are already similar struct defined in SpCore, but we choose to define separate
+// structs here for the following reasons. First, we avoid a dependency on RPCLib in SpCore. Second, the data
+// needed at the SpCore level is sometimes slightly different from the data needed at the SpEngine level, e.g.,
+// PropertyDesc needs to maintain a FProperty pointer but GameWorldServicePropertyDesc doesn't. Third, some
+// data types used in SpCore cannot be sent via RPCLib, e.g., FProperty* in PropertyDesc.
+
+MSGPACK_ADD_ENUM(EIncludeSuperFlag::Type);
+
+struct GameWorldServicePropertyDesc
+{
+    uint64_t property_;
+    uint64_t value_ptr_;
+
+    MSGPACK_DEFINE_MAP(property_, value_ptr_);
+};
+
 class GameWorldService {
 public:
     GameWorldService() = delete;
```

Comment:
No need for the assert on line 51, unless we're willing to do it consistently in all entry points.